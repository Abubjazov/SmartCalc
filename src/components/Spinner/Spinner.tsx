import { Helmet } from 'react-helmet'

export const Spinner = (): JSX.Element => {
	return (
		<svg
			className='spinner'
			width='200'
			height='200'
			viewBox='0 0 100 100'
			preserveAspectRatio='xMidYMid'
		>
			<Helmet>
				<title>Выполнение...</title>
			</Helmet>
			<path
				fill='none'
				stroke='#3b3b3b'
				strokeWidth='8'
				strokeDasharray='42.76482137044271 42.76482137044271'
				d='M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z'
				strokeLinecap='round'
			>
				<animate
					attributeName='stroke-dashoffset'
					repeatCount='indefinite'
					dur='1s'
					keyTimes='0;1'
					values='0;256.58892822265625'
				></animate>
			</path>
		</svg>
	)
}
