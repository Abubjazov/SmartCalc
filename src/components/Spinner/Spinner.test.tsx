import { render } from '@testing-library/react'

import { Spinner } from './Spinner'

describe('Component: Spinner', () => {
	test('should render Spinners', () => {
		const { asFragment } = render(<Spinner />)

		expect(asFragment()).toMatchSnapshot()
	})
})
