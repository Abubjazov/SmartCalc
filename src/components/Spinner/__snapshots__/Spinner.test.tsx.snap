// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Spinner should render Spinners 1`] = `
<DocumentFragment>
  <svg
    class="spinner"
    height="200"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
    width="200"
  >
    <path
      d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
      fill="none"
      stroke="#cccccc"
      stroke-dasharray="42.76482137044271 42.76482137044271"
      stroke-linecap="round"
      stroke-width="8"
    >
      <animate
        attributeName="stroke-dashoffset"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="0;256.58892822265625"
      />
    </path>
  </svg>
</DocumentFragment>
`;
